src/presentation/http/router/note.test.ts

```diff
@@ -385,6 +385,58 @@ describe('Note API', () => {
     test.todo('Returns 400 when parentId has incorrect characters and lenght');
   });
 
+  describe('DELETE /note/:notePublicId', () => {
+    test('Remove note', async () => {
+      const accessToken = global.auth(2);
+      const correctID = 'Hu8Gsm0sA1';
+
+      let response = await global.api?.fakeRequest({
+        method: 'DELETE',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.json()).toStrictEqual({ message: 'Note not found' });
+    });
+
+    test('Should remove all note relations contains note id', async () => {
+      const accessToken = global.auth(2);
+      const correctID = 'f43NU75weU';
+
+      let response = await global.api?.fakeRequest({
+        method: 'DELETE',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
+
+      expect(response?.statusCode).toBe(200);
+
+      response = await global.api?.fakeRequest({
+        method: 'GET',
+        headers: {
+          authorization: `Bearer ${accessToken}`,
+        },
+        url: `/note/${correctID}`,
+      });
```

Comment:
its not clear why what is happening here. Why do you check the same note?

Reply:
Because GET query returns both the note and its parent, so we can check if the relation was really deleted from the database `note_relations`

Reply:
where do you check that relation is removed?

Reply:
If there is no note relations, the GET request returns neither the note itself nor its parent, since they have been deleted

Reply:
right now it only checks that note itself was deleted, not the relation